### PR TITLE
Add vSphere static IP fields check

### PR DESCRIPTION
**Pull request by sgaoshang**
[OCP-67238 - [ipi-on-vsphere] static IPs field hosts check](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-67238)

### CODE REVIEW

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-vsphere-ipi-invaid-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-vsphere-ipi-invaid-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-vsphere-ipi-invaid-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-vsphere-ipi-invaid-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-vsphere-ipi-invaid-fields-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-vsphere-ipi-invaid-fields-f28

File: ci-operator/step-registry/cucushift/installer/check/vsphere/invalid-fields/cucushift-installer-check-vsphere-invalid-fields-commands.sh [hunk 1]
```diff
@@ -97,7 +96,210 @@ else
...
+CONFIG="${dir}/install-config.yaml"
+PATCH="${dir}/invalid-role-name.yaml.patch"
+cat >"${PATCH}" <<EOF
+platform:
+  vsphere:
+    hosts:
+      - role: bootstrap-invalid
+      - role: control-plane-invalid
+      - role: control-plane-invalid
+      - role: control-plane-invalid
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/vsphere/invalid-fields/cucushift-installer-check-vsphere-invalid-fields-commands.sh:155]: Shall we need the above 3 duplicated lines ?

Comment (sgaoshang) [reply ci-operator/step-registry/cucushift/installer/check/vsphere/invalid-fields/cucushift-installer-check-vsphere-invalid-fields-commands.sh:155]: We have static IP field like following, in order to change all role names I repeated them
```  
    hosts:
    - role: bootstrap
      networkDevice:
        ipAddrs:
        - 192.168.221.120/24
        gateway: 192.168.221.1
        nameservers:
        - 192.168.221.1
    - role: control-plane
      # failureDomain: us-east-1a
      networkDevice:
        ipAddrs:
        - 192.168.221.121/24
        gateway: 192.168.221.1
        nameservers:
        - 192.168.221.1
    - role: control-plane
      # failureDomain: us-east-1b
      networkDevice:
        ipAddrs:
        - 192.168.221.122/24
        gateway: 192.168.221.1
        nameservers:
        - 192.168.221.1
    - role: control-plane
      # failureDomain: us-east-1c
      networkDevice:
        ipAddrs:
        - 192.168.221.123/24
        gateway: 192.168.221.1
        nameservers:
        - 192.168.221.1
    - role: compute
      networkDevice:
        ipAddrs:
        - 192.168.221.124/24
        gateway: 192.168.221.1
        nameservers:
        - 192.168.221.1
    - role: compute
      networkDevice:
        ipAddrs:
        - 192.168.221.125/24
        gateway: 192.168.221.1
        nameservers:
        - 192.168.221.1
```

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/check/vsphere/invalid-fields/cucushift-installer-check-vsphere-invalid-fields-commands.sh:155]: got it.

**Pull request merged**